presto-kafka/src/main/java/io/prestosql/plugin/kafka/KafkaRecordSet.java

```diff
@@ -269,4 +278,53 @@ public void close()
             kafkaConsumer.close();
         }
     }
+
+    public static FieldValueProvider headerMapValueProvider(MapType varcharMapType, Headers headers)
```

Comment:
I'd assume empty headers case is not a rare case, so we could want to cache a result for this (empty block).

@dain wdyt?

Reply:
@0xE282B0 can you address this one unless the assumption about the fact, that empty headers is a common thing, is false?

Reply:
sure.

Reply:
Is there an easier way to get an empty MapBlock than creating a MapType and using a MapBlockBuilder?

```
MapType mapType = new MapType(VarcharType.VARCHAR, new ArrayType(VarbinaryType.VARBINARY),
        MethodHandles.empty(methodType(Boolean.class, Block.class, int.class, long.class)),
        MethodHandles.empty(methodType(Boolean.class, Block.class, int.class, Block.class, int.class)),
        MethodHandles.empty(methodType(long.class, Object.class)),
        MethodHandles.empty(methodType(long.class, Object.class)));
BlockBuilder mapBlockBuilder = new MapBlockBuilder(mapType, null, 0);
mapBlockBuilder.beginBlockEntry();
mapBlockBuilder.closeEntry();
Block emptyMapBlock = mapType.getObject(mapBlockBuilder, 0);
```

https://github.com/prestosql/presto/pull/4462/commits/414ddcb4a71e0c6d42a9a7ea62b105bcaec0e8fe#diff-1938e94844afe6ab2ee40936951490d9R300-R308

Reply:
I am not aware. Looks good to me. 

Nit: can you rename `createEmptyMapBlockProvider` to `createEmptyHeadersFieldProvider` 
and `EMPTY_MAP_BLOCK_PROVIDER` to `EMPTY_HEADERS_FIELD_PROVIDER`

Reply:
Also please add a test when we are reading `_headers` column but headers map is empty.

Reply:
Naming sounds better - more specific.
The empty header test doesn't look as straightforward as I wanted it to be, but it works. Since the `assertQuery` does not work on maps and VARBINARY does not work with JSON, I converted the map type and then cast it to JSON to compare it with empty objects. 
```
SELECT cast(transform_values(_headers,(k,v)->transform(v,x->from_utf8(x))) AS JSON)
```
Recomendations welcome 😉

Reply:
I think the test could be a bit more readable if headers topic also defined named `id` field. It would make assertions simpler.
Then if messages with empty headers has `id` equal to `1` and `2`.
The assertion would look like:
```
 assertQuery("SELECT id FROM default." + headersTopic + " WHERE cardinality(_headers) = 0",
                "VALUES (1), (2)");
```

The other test query could be more readable too if you used `id` in `WHERE` clause.

WDYT?

Reply:
Unfortunately we don't have an ID, but we can enumerate the messages based on the value of the message.
```
assertQuery("SELECT _message FROM default." + headersTopic + " WHERE cardinality(_headers) = 0",
               "VALUES ('1'),('2')");
```

Reply:
I was thinking of extending the schema with `id` but using `_message` is totally fine too.